fs/dafsclient/rmtclient.cpp

```diff
@@ -610,499 +627,492 @@ void clientAddSocketToCache(SocketEndpoint &ep,ISocket *socket)
 
 //---------------------------------------------------------------------------
 
-void CRemoteBase::connectSocket(SocketEndpoint &ep, unsigned localConnectTime, unsigned localRetries)
+void CRemoteBase::connectSocket(SocketEndpoint &ep, unsigned connectTimeoutMs, unsigned connectRetries)
 {
-	unsigned retries = 3;
-
-	if (localConnectTime)
-	{
-		if (localRetries)
-			retries = localRetries;
-		if (localConnectTime > maxConnectTime)
-			localConnectTime = maxConnectTime;
-	}
-	else
-		localConnectTime = maxConnectTime;
-
-	sRFTM tm(localConnectTime);
-
-	// called in CConnectionTable::crit
-
-	if (ep.equals(lastfailep)) {
-		if (msTick()-lastfailtime<DAFS_CONNECT_FAIL_RETRY_TIME) {
-			StringBuffer msg("Failed to connect (host marked down) to dafilesrv/daliservix on ");
-			ep.getUrlStr(msg);
-			throw createDafsException(DAFSERR_connection_failed,msg.str());
-		}
-		lastfailep.set(NULL);
-		retries = 1;    // on probation
-	}
-	while(retries--) {
-		CriticalUnblock unblock(CConnectionTable::crit); // allow others to connect
-		StringBuffer eps;
-		if (TF_TRACE_CLIENT_CONN) {
-			ep.getUrlStr(eps);
-			if (ep.port == securitySettings.daFileSrvSSLPort)
-				PROGLOG("Connecting SECURE to %s", eps.str());
-			else
-				PROGLOG("Connecting to %s", eps.str());
-			//PrintStackReport();
-		}
-		bool ok = true;
-		try {
-			if (tm.timemon) {
-				unsigned remaining;
-				if (tm.timemon->timedout(&remaining))
-					throwJSocketException(JSOCKERR_connection_failed);
-				socket.setown(ISocket::connect_timeout(ep,remaining));
-			}
-			else
-				socket.setown(ISocket::connect(ep));
-			if (ep.port == securitySettings.daFileSrvSSLPort)
-			{
+    if (!connectTimeoutMs)
+        connectTimeoutMs = dafsConnectTimeoutMs;
```

Comment:
NB: this is a slight (deliberate) change in semantics.
Previously if connectTimeoutMs was supplied and higher than the global default, it was effectively ignored.
After this change a supplied connectTimeout overrides the global default.

NB2: The only place a timeout was supplied it was (5000) it was always less than the hard-wired default, so this change does not effect any existing code.

Reply:
Is there a minimum on the connect time ?  Perhaps we shoud make sure it is at least some minimum ?

Reply:
Not sure about that. If caller wants a very low minimum that's up to them.

Reply:
connect_timeout() uses timeout passed in, connect_wait() ups timeout to >= 10 sec.